server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -601,11 +602,17 @@ public void apply(Settings value, Settings current, Settings previous) {
                 NodeDuressSettings.SETTING_CPU_THRESHOLD,
                 NodeDuressSettings.SETTING_HEAP_THRESHOLD,
                 SearchShardTaskSettings.SETTING_TOTAL_HEAP_PERCENT_THRESHOLD,
+                HeapUsageTracker.SETTING_HEAP_PERCENT_THRESHOLD_FOR_SEARCH_QUERY,
                 HeapUsageTracker.SETTING_HEAP_PERCENT_THRESHOLD,
```

Comment:
I think this would be a good time to rename the previous settings' _field_ names to indicate they belong to shard tasks. We can also drop 'FOR' in the field names.

```
HeapUsageTracker.SETTING_HEAP_PERCENT_THRESHOLD_SEARCH_TASK
HeapUsageTracker.SETTING_HEAP_PERCENT_THRESHOLD_SEARCH_SHARD_TASK  // renamed
```

User facing name for existing settings already indicate shard tasks, so no changes needed there that break backward-compatibility. (Eg. "search_backpressure.search_shard_task.heap_percent_threshold").

Reply:
Good point. Renamed the fields.